LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/ModuleConnection/ModuleConnectionHost.cs

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaWan.NetworkServer.BasicsStation.ModuleConnection
+{
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Client.Exceptions;
+    using Microsoft.Azure.Devices.Shared;
+    using Microsoft.Extensions.Logging;
+    using System;
+    using System.Configuration;
+    using System.Net;
+    using System.Text.Json;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public sealed class ModuleConnectionHost : IAsyncDisposable
+    {
+        private readonly NetworkServerConfiguration networkServerConfiguration;
+        private readonly IClassCDeviceMessageSender classCMessageSender;
+        private readonly ILoRaDeviceRegistry loRaDeviceRegistry;
+        private readonly ILoraModuleClient loRaModuleClient;
+
+        public ModuleConnectionHost(
+            NetworkServerConfiguration networkServerConfiguration,
+            IClassCDeviceMessageSender defaultClassCDevicesMessageSender,
+            ILoRaDeviceRegistry loRaDeviceRegistry)
+        {
+            this.networkServerConfiguration = networkServerConfiguration ?? throw new ArgumentNullException(nameof(networkServerConfiguration));
+            this.classCMessageSender = defaultClassCDevicesMessageSender ?? throw new ArgumentNullException(nameof(defaultClassCDevicesMessageSender));
+            this.loRaDeviceRegistry = loRaDeviceRegistry ?? throw new ArgumentNullException(nameof(loRaDeviceRegistry));
+            this.loRaModuleClient = new LoRaModuleClient();
+        }
```

Comment:
Constructors should use chaining to avoid duplication:

```suggestion
        public ModuleConnectionHost(
            NetworkServerConfiguration networkServerConfiguration,
            IClassCDeviceMessageSender defaultClassCDevicesMessageSender,
            ILoRaDeviceRegistry loRaDeviceRegistry) :
            this(networkServerConfiguration, defaultClassCDevicesMessageSender, loRaDeviceRegistry, new LoRaModuleClient())
        {
        }
```